### ISSUE TITLE
Editor integration

### ISSUE THREAD

**Issue by athas** [shwestrick/smlfmt#90]
I don't know whether this should be a PR adding it to the README or whatnot, but I have found this Emacs function quite handy for running `smlfmt` on the current file buffer:

```elisp
 (defun smlfmt ()
   (interactive)
   (save-buffer)
   (let* ((file (buffer-file-name))
          (ret (with-temp-buffer
                 (let ((code (call-process "smlfmt" nil '(t t) nil "--force" file)))
                   (if (= code 0)
                       nil
                     (buffer-string))))))
     (if (eq ret nil)
         (revert-buffer t t t)
       (message "%s" ret))))
```

**Comment by shwestrick** [comment]
Maybe we could create something like:
```
<root>
...
├── editor-integration/
│   └── emacs/
...     ├── README
        └── ...
```

And, we could add it to the feature list in the top-level README.

If you put together a PR I'm happy to merge it in!

**Comment by athas** [comment]
I will, but I found an even better way to write this (apparently there's an Emacs package specifically for integrating autoformatters).

**Comment by shwestrick** [comment]
sounds great!

**Comment by athas** [comment]
I am building it here: https://github.com/diku-dk/smlfmt.el

It is its own repository, because that is what the Emacs package repositories prefer.

It requires the newest (unreleased) smlfmt, so I will not submit it to the repositories before the next smlfmt release.

**Comment by shwestrick** [comment]
Just released [v1.1.0](https://github.com/shwestrick/smlfmt/releases/tag/v1.1.0), and added a mention of `smlfmt.el` to the feature list in the README.

**Comment by athas** [comment]
Great!  [I have submitted smlfmt.el to MELPA](https://github.com/melpa/melpa/pull/8805) and with some luck they will accept it.
